migration-tools/migrate-rosdistro.py

```diff
@@ -183,20 +183,24 @@ def write_tracks_file(tracks, commit_msg=None):
         # interactivity and :{auto} may result in a previously unreleased tag
         # on the development branch being released for the first time.
         if dest_track['version'] in [':{ask}', ':{auto}']:
-            # Override the version for this release to guarantee the same version is released.
+            # Override the version for this release to guarantee the same version from our
+            # source distribution is released.
             dest_track['version_saved'] = dest_track['version']
-            dest_track['version'] = dest_track['last_version']
-            write_tracks_file(tracks, f'Update {args.dest} track to release exactly last-released version.')
+            source_version, source_inc = source_distribution.repositories[repo_name].release_repository.version.split('-')
+            dest_track['version'] = source_version
+            write_tracks_file(tracks, f'Update {args.dest} track to release the same version as the source distribution.')
 
         if dest_track['release_tag'] == ':{ask}' and 'last_release' in dest_track:
             # Override the version for this release to guarantee the same version is released.
             dest_track['release_tag_saved'] = dest_track['release_tag']
             dest_track['release_tag'] = dest_track['last_release']
             write_tracks_file(tracks, f'Update {args.dest} track to release exactly last-released tag.')
 
+        # Update release increment for the upcoming release.
+        release_inc = str(int(source_inc) + 1)
```

Comment:
Rather than overwriting the tag, I'd opt for a release inc which is "newer than both". It should have the same effect without any force pushing.

```suggestion
        release_inc = str(max(int(source_inc), int(dest_track['release_inc'])) + 1)
```

Reply:
+1 in case there was a missed push somewhere.

Reply:
There are still cases this solution wouldn't cover with multiple uncommitted releases.

Using ros2_control as an example again. If our source distribution was at version 2.1.0-1 and the release repository contained tagged but unpublished in rosdistro releases 2.1.0-2 and 2.2.0-1, then the migration tool would see the last release `2.2.0` and release increment `"1"` in the tracks file and create a release `2.1.0-2` from
```python
    release_inc = str(max(int("1"), int("1")) + 1)
```
which would clobber the upstream 2.1.0-2.

This is what led me to think that the only reliable way to prevent clobbering unreleased tags is to scan for all tags matching the target release version to pick a strictly greater release increment.

Reply:
I see - good catch. Well it's still an improvement, no?

Reply:
Without a doubt.

The question hidden in my comment was Should we take the improvement and continue the migration or push for a solution that handles the further edge case?

Reply:
I don't think that there's any current repositories with that specific scenario so I would be inclined to take your improvement and proceed with the migration. But I'll add a comment documenting the additional edge case and try to revisit it before we pull this out again for Humble.

Reply:
I agree with your analysis - we can't do better without scanning the tags, which is a bit more work and IMO sort-of reaches further into the repositories than I think the migration tooling should.

+1 to move forward.

Reply:
Updated in d80eb24.